File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int i=0,j=0,max_window=-1;
        while(j<nums.size())
        {
            if(nums[j] == 0)
            {
                k--;
            }
            while(k == -1)
            {
                if(nums[i] == 0)
                {
                    k++;
                }
                i++;
            }
            int current_window = (j-i+1);
            max_window = max(current_window,max_window);
            j++;
        }
        return max_window;
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        i = 0
        best = 0
        count = 0
        for j in range(len(nums)):
            if nums[j] == 0:
                count += 1
            while count > k:
                if nums[i] == 0:
                    count -= 1
                i += 1
            best = max(best, j - i + 1)
        return best
